LongestPole/Main.lean

```diff
@@ -0,0 +1,185 @@
+/-
+Copyright (c) 2023 Scott Morrison. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Scott Morrison
+-/
+import ImportGraph
+import Mathlib.Data.String.Defs
+import Std.Lean.Util.Path
+import Cli
+
+/-!
+# `lake exe pole`
+
+Longest pole analysis for Mathlib build times.
+-/
+
+set_option autoImplicit true
+
+open Cli
+
+open Lean Meta
+
+open Lean Core System
+
+/-- Runs a terminal command and retrieves its output -/
+def runCmd (cmd : String) (args : Array String) (throwFailure := true) : IO String := do
+  let out ← IO.Process.output { cmd := cmd, args := args }
+  if out.exitCode != 0 && throwFailure then throw $ IO.userError out.stderr
+  else return out.stdout
+
+def runCurl (args : Array String) (throwFailure := true) : IO String := do
+  runCmd "curl" args throwFailure
+
+def speedCenterRunJson (hash : String) : IO String :=
+  -- e7b27246-a3e6-496a-b552-ff4b45c7236e is the `repo_id` for the `mathlib4` repository.
+  runCurl #["http://speed.lean-fro.org/mathlib4/api/run/e7b27246-a3e6-496a-b552-ff4b45c7236e?hash=" ++ hash]
+
+structure SpeedCenterSource_1 where
```

Comment:
I think this should be called `CommitSource` or similar, since it corresponds to `type: "COMMIT"`

Reply:
@BoltonBailey, did you intend to apply this change?

Reply:
Yes, and I did so.